timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/HeldLocks.java

```diff
@@ -79,6 +85,7 @@ public synchronized boolean unlock() {
 
         for (AsyncLock lock : acquiredLocks) {
             lock.unlock(token.getRequestId());
+            lockEventProcessor.registerUnlock(lock.getDescriptor());
```

Comment:
I think this is a bit weird with the targeted sweep lock (unlocks are async) + immutable timestamp lock (why would you want to watch this lock)? which further reinforces my belief that this should exist inside `ExclusiveLock`

Reply:
Conceptually that makes sense, and I think it's fine to put it there. I was initially trying to avoid touching `ExclusiveLock` completely